packages/uniforms/__suites__/DateField.tsx

```diff
@@ -1,106 +1,245 @@
     renderWithZod({
-      element: <DateField name="x" type="date" />,
-      onChange,
+      element: <DateField name="x" id="y" data-testid="z" />,
       schema: z.object({ x: z.date() }),
     });
-
-    const input = screen.getByLabelText(/^X/);
-    expect(input).toHaveValue('');
-
-    await userEvent.type(input, date);
-    expect(onChange).toHaveBeenLastCalledWith('x', new Date(`${date}Z`));
+    expect(findClosestInputWithTestId('z')?.id).toBe('y');
   });
 
-  test('<DateField> - handles "date" type correctly (overflow)', async () => {
-    const date = '12345-06-07';
-    const onChange = jest.fn();
+  test('<DateField> - renders a input with correct name', () => {
     renderWithZod({
-      element: <DateField name="x" type="date" />,
-      onChange,
+      element: <DateField name="x" data-testid="z" />,
       schema: z.object({ x: z.date() }),
     });
+    expect(findClosestInputWithTestId('z')?.name).toBe('x');
+  });
 
-    const input = screen.getByLabelText(/^X/);
-    expect(input).toHaveValue('');
-
-    await userEvent.type(input, date);
-    expect(onChange).not.toHaveBeenCalled();
+  test('<DateField> - renders an input with correct disabled state', () => {
+    renderWithZod({
+      element: <DateField name="x" disabled data-testid="z" />,
+      schema: z.object({ x: z.date() }),
+    });
+    expect(findClosestInputWithTestId('z')?.disabled).toBe(true);
   });
 
-  test('<DateField> - handles "datetime-local" type correctly (empty)', async () => {
-    const date = '2021-01-01T11:11';
-    const onChange = jest.fn();
+  test('<DateField> - renders a input with correct label (specified)', () => {
     renderWithZod({
-      element: <DateField name="x" />,
-      model: { x: new Date(`${date}Z`) },
-      onChange,
+      element: <DateField name="x" label="DateFieldLabel" />,
       schema: z.object({ x: z.date() }),
     });
+    expect(screen.getByText('DateFieldLabel')).toBeInTheDocument();
+  });
 
-    const input = screen.getByLabelText(/^X/);
-    expect(input).toHaveValue(date);
+  test('<DateField> - renders a input with correct value (default)', () => {
+    renderWithZod({
+      element: <DateField name="x" data-testid="z" />,
+      schema: z.object({ x: z.date() }),
+    });
+    const input = findClosestInputWithTestId('z');
+    expect(input).toBeInTheDocument();
+    expect(input?.value).toBe('');
+  });
 
-    await userEvent.clear(input);
-    expect(onChange).toHaveBeenLastCalledWith('x', undefined);
+  test('<DateField> - renders a input with correct value (specified)', () => {
+    const now = new Date('2024-01-01 12:00:00');
+    renderWithZod({
+      element: <DateField name="x" data-testid="z" value={now} />,
+      schema: z.object({ x: z.date() }),
+    });
+    const input = findClosestInputWithTestId('z');
+    expect(input).toBeInTheDocument();
+    expect(input!.value).toBe(getDateString(now, skipTestAntD));
```

Comment:
I get the idea, but it looks wrong. Even if you will use `options.theme === 'antd'` in the future, it won't be readable enough. Create a variable above:
```
const useISOFormat = options.theme === 'antd'
```

Maybe even put it above tests to avoid duplication (there are two places to refactor).

Reply:
Changed ✅